src/System.Net.Security/src/System/Net/Security/SslState.cs

```diff
@@ -1373,15 +1373,24 @@ private void HandleQueuedCallback(ref object queuedStateRequest)
                 return;
             }
             queuedStateRequest = null;
-                        
+
             switch (obj)
             {
                 case LazyAsyncResult lazy:
                     lazy.InvokeCallback();
                     break;
                 case TaskCompletionSource<int> taskCompletionSource when taskCompletionSource.Task.AsyncState != null:
                     Memory<byte> array = (Memory<byte>)taskCompletionSource.Task.AsyncState;
-                    taskCompletionSource.SetResult(CheckOldKeyDecryptedData(array));
+                    int oldKeyResult = -1;
+                    try
+                    {
+                        oldKeyResult = CheckOldKeyDecryptedData(array);
+                    }
+                    catch (Exception exc)
+                    {
+                        taskCompletionSource.SetException(exc);
```

Comment:
You need to break here. Otherwise this will crash when it tries to Set result after.